tests/e2e/concurrent-e2e/concurrent_test.go

```diff
@@ -0,0 +1,346 @@
+						Duration: &pipelineSpecSourceDuration,
+					},
+				},
+				Scale: numaflowv1.Scale{Min: &fourPods, Max: &fivePods, ZeroReplicaSleepSeconds: &zeroReplicaSleepSec},
+			},
+			{
+				Name: "out",
+				Sink: &numaflowv1.Sink{
+					AbstractSink: numaflowv1.AbstractSink{
+						Log: &numaflowv1.Log{},
+					},
+				},
+				Scale: numaflowv1.Scale{ZeroReplicaSleepSeconds: &zeroReplicaSleepSec},
+			},
+		},
+		Edges: []numaflowv1.Edge{
+			{
+				From: sourceVertexName,
+				To:   "out",
+			},
+		},
+	}
+
+	dataLossPipelineSpec = numaflowv1.PipelineSpec{
+		InterStepBufferServiceName: isbServiceRolloutName,
+		Lifecycle: numaflowv1.Lifecycle{
+			PauseGracePeriodSeconds: ptr.To(int64(120)),
+		},
+		Vertices: []numaflowv1.AbstractVertex{
+			{
+				Name: sourceVertexName,
+				Source: &numaflowv1.Source{
+					Generator: &numaflowv1.GeneratorSource{
+						RPU:      &pipelineSpecSourceRPU,
+						Duration: &pipelineSpecSourceDuration,
+					},
+				},
+				Scale: numaflowv1.Scale{Min: &fourPods, Max: &fivePods, ZeroReplicaSleepSeconds: &zeroReplicaSleepSec},
+			},
+			{
+				Name: "cat",
+				UDF: &numaflowv1.UDF{
+					Builtin: &numaflowv1.Function{
+						Name: "cat",
+					},
+				},
+				Scale: numaflowv1.Scale{Min: &onePod, Max: &onePod, ZeroReplicaSleepSeconds: &zeroReplicaSleepSec},
+			},
+			{
+				Name: "out",
+				Sink: &numaflowv1.Sink{
+					AbstractSink: numaflowv1.AbstractSink{
+						Log: &numaflowv1.Log{},
+					},
+				},
+				Scale: numaflowv1.Scale{Min: &onePod, Max: &onePod, ZeroReplicaSleepSeconds: &zeroReplicaSleepSec},
+			},
+		},
+		Edges: []numaflowv1.Edge{
+			{
+				From: sourceVertexName,
+				To:   "cat",
+			},
+			{
+				From: "cat",
+				To:   "out",
+			},
+		},
+	}
+
+	monoVertexScaleMin = int32(4)
+	monoVertexScaleMax = int32(5)
+
+	initialMonoVertexSpec = numaflowv1.MonoVertexSpec{
+		Scale: numaflowv1.Scale{
+			Min: &monoVertexScaleMin,
+			Max: &monoVertexScaleMax,
+		},
+		Source: &numaflowv1.Source{
+			UDSource: &numaflowv1.UDSource{
+				Container: &numaflowv1.Container{
+					Image: "quay.io/numaio/numaflow-go/source-simple-source:stable",
+				},
+			},
+		},
+		Sink: &numaflowv1.Sink{
+			AbstractSink: numaflowv1.AbstractSink{
+				Blackhole: &numaflowv1.Blackhole{},
+			},
+		},
+	}
+
+	volSize, _     = apiresource.ParseQuantity("10Mi")
+	isbServiceSpec = numaflowv1.InterStepBufferServiceSpec{
+		Redis: nil,
+		JetStream: &numaflowv1.JetStreamBufferService{
+			Version: initialJetstreamVersion,
+			Persistence: &numaflowv1.PersistenceStrategy{
+				VolumeSize: &volSize,
+			},
+		},
+	}
+
+	dataLossISBServiceSpec = numaflowv1.InterStepBufferServiceSpec{
+		Redis: nil,
+		JetStream: &numaflowv1.JetStreamBufferService{
+			Version: updatedJetstreamVersion,
+			Persistence: &numaflowv1.PersistenceStrategy{
+				VolumeSize: &volSize,
+			},
+		},
+	}
+
+	updatedMemLimit, _            = apiresource.ParseQuantity("2Gi")
+	ISBServiceSpecNoDataLossField = numaflowv1.InterStepBufferServiceSpec{
```

Comment:
nit: could we have consistency for naming of these variables, with the `dataLossISBServiceSpec` above?

Reply:
(same for `ISBServiceSpecRecreateField` below)